src/components/TextArea.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FunctionComponent } from "react";
+import { TextAreaProps as AntTextAreaProps } from "antd/lib/input";
+import { Input } from "antd";
+import { FormItem } from "@components/FormItem";
+import { FormItemProps } from "antd/lib/form";
+import { classNames } from "@utils/format.utils";
+
+const { TextArea: AntTextArea } = Input;
+const DEFAULT_TEXTAREA_ROWS = 6;
+
+interface TextAreaProps {
+	rows?: AntTextAreaProps["rows"];
+	placeholder?: AntTextAreaProps["placeholder"];
+	value?: AntTextAreaProps["value"];
+	className?: AntTextAreaProps["className"];
+	label: string;
+	/**
+	 * Controls wheter the input can be interacted with or not.
+	 * This differs from setting `disabled` to `true` in that by
+	 * setting this option to true will remove any styling from the
+	 * input that makes it look like an input.
+	 */
+	isDisplay?: boolean;
+	/**
+	 * Specifis whether the input has validation feedback
+	 */
+	hasFeedback?: FormItemProps["hasFeedback"];
+	/**
+	 * The prompt message. If not provided, the prompt message will be generated by the validation rule.
+	 */
+	help?: FormItemProps["help"];
+	/**
+	 * The result of the validation rules
+	 */
+	validateStatus?: "error" | "";
+	disabled?: boolean;
+	required?: boolean;
+	/**
+	 * Optional description to provide more information about the input
+	 */
+	description?: string;
+	/**
+	 * Form item name
+	 */
+	name?: FormItemProps["name"];
+}
+
+export const TextArea: FunctionComponent<TextAreaProps> = ({
+	rows = DEFAULT_TEXTAREA_ROWS,
+	placeholder,
+	className,
+	label,
+	isDisplay,
+	value,
+	hasFeedback,
+	validateStatus,
+	disabled,
+	description,
+	name,
+	help,
+}) => {
+	className = classNames([
```

Comment:
what is the main benefit over just using a template string?

Reply:
Say you have a template string like this: 
```typescript
const somethingThatIsFalse = false;
const className = `${somethingThatIsFalse && "bg-red-500"} text-white`;
```

The resulting string would be `"false text-white"`.

The same with optional `className` props:
```typescript
const className = `text-white ${className}`;
```

If no `className` prop was given, the resulting string would be `"text-white "` (notice the space at the end)

Adding them all to an array, filtering out falsy values and joining them with a space results in a clean string.